gcloud-java-storage/src/main/java/com/google/gcloud/storage/Blob.java

```diff
@@ -118,7 +117,19 @@ public Blob(Storage storage, BlobInfo info) {
    */
   public Blob(Storage storage, String bucket, String blob) {
     this.storage = checkNotNull(storage);
-    this.info = BlobInfo.of(checkNotNull(bucket), checkNotNull(blob));
+    this.info = BlobInfo.builder(BlobId.of(bucket, blob)).build();
```

Comment:
I agree with your reasons but I would not do this in the constructor. If we go this way I'd suggest to have a single constructor `Blob(BlobInfo info)` and two factory methods (name is arbitrary):
```java
Blob.fetch(BlobId id);
Blob.fetch(String bucket, String name);
```
Only these two factories do the remote get and instantiate a `Blob` object accordingly. The user can still use the constructor and instantiate a `Blob` with any `BlobInfo`.